src/Tizen.MachineLearning.Train/Tizen.MachineLearning.Train/Dataset.cs

```diff
@@ -0,0 +1,145 @@
+        /// </remarks>
+        ~Dataset()
+        {
+            Dispose(false);
+        }
+
+        /// <summary>
+        /// Releases any unmanaged resources used by this object.
+        /// </summary>
+        /// <since_tizen> 10 </since_tizen>
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        /// <summary>
+        /// Releases any unmanaged resources used by this object including opened handle.
+        /// </summary>
+        /// <param name="disposing">If true, disposes any disposable objects. If false, does not dispose disposable objects.</param>
+        /// <since_tizen> 10 </since_tizen>
+        protected virtual void Dispose(bool disposing)
+        {
+            if (disposed)
+                return;
+            if (disposing)
+            {
+                // release managed object
+            }
+            // release unmanaged object
+            if (handle != IntPtr.Zero)
+            {
+                // Destroy dataset.
+                NNTrainerError ret = Interop.Dataset.Destroy(handle);
+                NNTrainer.CheckException(ret, "Failed to destroy dataset instance");
+                Log.Info(NNTrainer.Tag, "Destroy Dataset");
+
+                handle = IntPtr.Zero;
+            }
+            disposed = true;
+        }
+
+        /// <summary>
+        /// Adds data file to dataset.
+        /// </summary>
+        /// <remarks>
+        /// Use this function to add a data file from where data is retrieved.
+        /// If you want to access only internal storage by using this function,
+        /// you should add privilege %http://tizen.org/privilege/mediastorage. Or, if you
+        /// want to access only external storage by using this function, you should add
+        /// privilege %http://tizen.org/privilege/externalstorage. If you can access both
+        /// storage, you must add all privilege
+        /// </remarks>
+        /// <param name="mode">The phase where this generator should be used.</param>
+        /// <param name="file">file path.</param>
+        /// <since_tizen> 10 </since_tizen>
+        public void AddFile(NNTrainerDatasetMode mode, string file)
+        {
+            if (string.IsNullOrEmpty(file))
+                NNTrainer.CheckException(NNTrainerError.InvalidParameter, "file is null");
+
+            NNTrainerError ret = Interop.Dataset.AddFile(handle, mode, file);
+            NNTrainer.CheckException(ret, "Failed to add file");
+            Log.Info(NNTrainer.Tag, $"Add file{file}");
+        }
+
+        internal IntPtr GetHandle()
+        {
+            return handle;
+        }
+
+        /// <summary>
+        /// Sets the neural network dataset property.
+        /// </summary>
+        /// <remarks>
+        /// Use this function to set dataset property for a specific mode.
+        /// The input format of property must be 'key = value' format.
```

Comment:
and what if it not in that format? would be ignored, threated as a flag (option without value), and I've noticed there is no exceptions in documentation